### PR TITLE
swagger api and few fixes

### PR DESCRIPTION
**Pull request by dambaev**
This PR introduces:
1. Swagger UI tool accessible from the second Documentation button from the Preview page, which provides an ability to perform API calls with curl using this UI;
2. Swagger JSON file accessible from  GET `/api/v1/swagger.json`;
3. reference to API specification and  https://github.com/dambaev/op-energy-api tool in the README;
4. few fixes that I came up by testing API as well.

The purpose of this PR is to rely on Swagger specification for API instead of reusing mempool's method of API documentation, which adds too much of manual work to keep documentation in-sync with implementation.

It is possible in the future to use `swagger-generator` to generate client API library right from API specification file (I am testing this feature currently, but haven't finished tests to say that it is ready to be used)

### CODE REVIEW

**Comment by dambaev** [comment]
currently dev instance is running this branch so you can test it online: http://dev-exchange.op-energy.info/oe/docs

File: backend/src/oe/api/op-energy.service.ts [hunk 1]
```diff
@@ -68,6 +73,25 @@ export class OpEnergyApiService {
       'value': num,
     };
   }
+
+  public verifyAccountSecret( rawString: string): AccountSecret {
+    if( rawString.length !== 64) {
+      throw new Error('verifyAccountSecret: length');
+    }
+    if( rawString[10] !== '0'
```
Comment (Natsvora) [inline backend/src/oe/api/op-energy.service.ts]: We should use enum for static values like '0', 'e'  etc. If possible please move this to enum

File: backend/src/oe/api/op-energy.service.ts [hunk 2]
```diff
@@ -436,14 +476,50 @@ export class OpEnergyApiService {
...
     // using our own block cache goes here
     return await bitcoinApi.$getBlock(hash.value);
   }
 
+  public async $registerNewUser( UUID: string): Promise<[ AccountSecret, AccountToken]> {
+    const rnd = [...Array(10)].map( _ => String.fromCharCode(Math.floor(Math.random() * 255))).join('');
+    var newHashArr = [ ... crypto.HmacSHA256(rnd, config.DATABASE.SECRET_SALT).toString().slice(0, 64)];
+    // secret should have special bytes in certain places to pass input verification
+    newHashArr[10] = '0';
+    newHashArr[30] = 'e';
```
Comment (Natsvora) [inline backend/src/oe/api/op-energy.service.ts]: Use same enum created above here

File: backend/src/oe/api/websocket.ts [hunk 3]
```diff
@@ -160,15 +160,15 @@ class OpEnergyWebsocket {
 
   // sends notifications to all the WebSockets' clients about new TimeStrike value
   handleNewTimeStrike(timeStrike: TimeStrike) {
-    if (!this.wss) {
+    if (OpEnergyWebsocket.wss === undefined) {
```
Comment (Natsvora) [inline backend/src/oe/api/websocket.ts]: Can't we use !OpEnergyWebsocket.wss here?

File: frontend/src/app/oe/components/oe-docs/oe-docs.component.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    @Inject(DOCUMENT) public document: Document,
+    public stateService: StateService,
+    private languageService: LanguageService,
+    private enterpriseService: EnterpriseService,
+    private httpClient: HttpClient,
+  ) {
+    this.baseUrl = document.location.protocol + '//' + document.location.host;
+    this.basePath = ''; // assume mainnet by default
+    this.stateService.networkChanged$.subscribe((network) => {
+      if (network === 'bisq') {
```
Comment (Natsvora) [inline frontend/src/app/oe/components/oe-docs/oe-docs.component.ts:50]: Move network name bisq into enum and use it

**Review by Natsvora — APPROVED** [review]

File: backend/src/oe/api/op-energy.service.ts (hunk 1 above)
Comment (dambaev) [reply backend/src/oe/api/op-energy.service.ts]: so I am trying to follow your idea and: enum can contain only values of simple types, so I can't keep, for example value of type [number, string] to be able to redefine both position and characters at such positions.

So instead I had refactored to definition of 2 arrays of type [number,string][] such that now we can keep value [10, '0'] and use forEach/reduce functions in creation/validation operations.

**Pull request merged**